packages/opentelemetry-resources/src/platform/node/detectors/AwsBeanstalkDetector.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Resource } from '../../../Resource';
+import { SERVICE_RESOURCE } from '../../../constants';
+import { Detector } from '../../../types';
+import { ResourceDetectionConfigWithLogger } from '../../../config';
+import * as fs from 'fs';
+
+/**
+ * The AwsBeanstalkDetector can be used to detect if a process is running in AWS Elastic
+ * Beanstalk and return a {@link Resource} populated with data about the beanstalk
+ * plugins of AWS X-Ray. Returns an empty Resource if detection fails.
+ *
+ * See https://docs.amazonaws.cn/en_us/xray/latest/devguide/xray-guide.pdf
+ * for more details about detecting information of Elastic Beanstalk plugins
+ */
+class AwsBeanstalkDetector implements Detector {
+  readonly BEANSTALK_CONF_PATH = '/var/elasticbeanstalk/xray/environment.conf';
+
+  async detect(config: ResourceDetectionConfigWithLogger): Promise<Resource> {
+    try {
+      const rawData = fs.readFileSync(this.BEANSTALK_CONF_PATH, 'utf8');
```

Comment:
No need to use sync method in an async context

Reply:
Thank you for your comment! Already fixed it and made a new commit!